src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java

```diff
@@ -296,4 +370,48 @@ public List<LimitViolationType> getLimitTypes(UUID resultUuid) {
     public List<com.powsybl.shortcircuit.Fault.FaultType> getFaultTypes(UUID resultUuid) {
         return resultService.findFaultTypes(resultUuid);
     }
+
+    public Optional<ShortCircuitParametersInfos> getParameters(final UUID parametersUuid) {
+        return parametersRepository.findById(parametersUuid).map(ShortCircuitService::fromEntity);
+    }
+
+    @Transactional
+    public boolean deleteParameters(final UUID parametersUuid) {
+        final boolean result = parametersRepository.existsById(parametersUuid);
+        if (result) {
+            parametersRepository.deleteById(parametersUuid);
+        }
+        return result;
+    }
+
+    @Transactional
+    public Optional<UUID> duplicateParameters(UUID sourceParametersUuid) {
+        return parametersRepository.findById(sourceParametersUuid)
+                                   .map(entity -> {
+                                       entityManager.detach(entity);
+                                       entity.setId(null);
+                                       return entity;
+                                   })
+                                   .map(parametersRepository::save)
+                                   .map(AnalysisParametersEntity::getId);
+    }
+
+    public UUID createParameters(@Nullable final ShortCircuitParametersInfos parameters) {
+        return parametersRepository.save(parameters != null ? toEntity(parameters) : new AnalysisParametersEntity()).getId();
+    }
+
+    /**
+     * Use the ability of Jackson to update an object with a partial Json.
+     */
+    @Transactional
+    public boolean updateOrResetParameters(final UUID parametersUuid, @Nullable final ShortCircuitParametersInfos givenParameters) {
+        AnalysisParametersEntity parameters = parametersRepository.findById(parametersUuid).orElse(null);
+        if (parameters == null) {
+            return false;
+        } else {
+            AnalysisParametersEntity newParameters = givenParameters != null ? toEntity(givenParameters) : new AnalysisParametersEntity();
+            entityManager.merge(newParameters.setId(parameters.getId()));
+            return true;
+        }
```

Comment:
Why not but if so, can you test this feature extensively so that we're sure the partial update is done properly ?

Reply:
I can add tests if you want, but do we need to test extensively code of Spring & hibernate that are already tests in their respective repositories?

It's not a partial update here as we build/convert a full entity to "merge", here we replace in fact, without having to do a function to `dst.setXXX(src.getXXX())` on each fields.